src/clients/java/src/client.zig

```diff
@@ -110,21 +115,59 @@ const ReflectionHelper = struct {
         assert(env.hasPendingException());
     }
 
-    pub fn buffer(env: *jui.JNIEnv, this_obj: jui.jobject) ?[]u8 {
+    pub fn get_send_buffer(env: *jui.JNIEnv, this_obj: jui.jobject) ?[]u8 {
         assert(this_obj != null);
-        assert(request_buffer_field_id != null);
-        assert(request_buffer_len_field_id != null);
+        assert(request_send_buffer_field_id != null);
+        assert(request_send_buffer_len_field_id != null);
 
-        var buffer_obj = env.getField(.object, this_obj, request_buffer_field_id) orelse return null;
+        var buffer_obj = env.getField(.object, this_obj, request_send_buffer_field_id) orelse return null;
         defer env.deleteReference(.local, buffer_obj);
 
-        var buffer_len = env.getField(.long, this_obj, request_buffer_len_field_id);
+        var buffer_len = env.getField(.long, this_obj, request_send_buffer_len_field_id);
         var address = env.getDirectBufferAddress(buffer_obj);
```

Comment:
Let's also use `getDirectBufferCapacity` to sanity check that we do in fact have at least `buffer_len` bytes.

Reply:
`env.getDirectBufferAddress` already calls `getDirectBufferCapacity` to return a slice.
We could explicitly `assert(buffer_len <= address.len)` but Zig does it when we take `address[0..buffer_len]`.

I think the best thing to do is that:
```suggestion
        var buffer: []u8 = env.getDirectBufferAddress(buffer_obj);
```